force-app/main/default/classes/SOQLDataProvider.cls

```diff
@@ -76,7 +76,7 @@ public inherited sharing virtual class SOQLDataProvider extends ChartDataProvide
     aChartData.detail = new List<Object>();
 
     for (AggregateResult aResult : Database.query(this.query)) {
-      aChartData.labels.add((String) aResult.get(LABEL_ALIAS));
+      aChartData.labels.add('' + aResult.get(LABEL_ALIAS));
```

Comment:
@scolladon what about using String.valueOf(get(LABEL_ALIAS)) ? Have you tried it? Not sure if it works fine for dates